### PR TITLE
Fix staging using inline methods in quotes

### PR DESCRIPTION
**Pull request by nicolasstucki**
Follow-up of #10781.

### CODE REVIEW

**Comment by nicolasstucki** [comment]
This includes a partial version of the code in #9984 that is enough for the use-case in staging.

File: compiler/src/dotty/tools/dotc/transform/Inlining.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import scala.collection.mutable
+import dotty.tools.dotc.core.Annotations._
+import dotty.tools.dotc.core.Names._
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.StagingContext._
+import dotty.tools.dotc.quoted._
+import dotty.tools.dotc.transform.TreeMapWithStages._
+import dotty.tools.dotc.typer.Inliner
+import dotty.tools.dotc.typer.ImportInfo.withRootImports
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Inlining.scala:28]: This import seems to be not used.

File: compiler/src/dotty/tools/dotc/transform/Inlining.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import Decorators._
+import Flags._
+import Types._
+import Contexts._
+import Symbols._
+import Constants._
+import ast.Trees._
+import ast.{TreeTypeMap, untpd}
+import util.Spans._
+import tasty.TreePickler.Hole
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Inlining.scala:14]: Import not used

File: compiler/src/dotty/tools/dotc/transform/Inlining.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import SymUtils._
+import NameKinds._
+import dotty.tools.dotc.ast.tpd
+import typer.Implicits.SearchFailureType
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Inlining.scala:18]: Import not used.

File: compiler/src/dotty/tools/dotc/transform/Inlining.scala [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+import scala.collection.mutable
+import dotty.tools.dotc.core.Annotations._
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Inlining.scala:21]: The two imports are not used.

**Review by liufengyun — APPROVED** [review]
LGTM

It seems `tests/disabled/run-staging/i3876-e.scala` can be enabled as well.

File: compiler/src/dotty/tools/dotc/transform/Inlining.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Inlining.scala:28]: Those are from  #9984. I kept them to simplify the merge afterward.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Inlining.scala:28]: I will cleanup all imports in #9984

**Pull request merged**